job/service.go

```diff
@@ -760,6 +804,21 @@ func (srv *Service) fetchSpecsForGivenJobNames(ctx context.Context, projectSpec
 	return jobSpecs, nil
 }
 
+func (srv *Service) IsJobDestinationDuplicate(ctx context.Context, jobSpec models.JobSpec) (string, error) {
+	jobWithSameDestination, err := srv.jobSpecRepository.GetJobByResourceDestination(ctx, jobSpec.ResourceDestination)
```

Comment:
looks like this will be useful if the user is using local job spec? since if using server job spec, it will always return the first record encountered in db, and will not detect the existing duplicated job destination in the system.

Reply:
have modified the implementation